pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go

```diff
@@ -225,23 +227,39 @@ func (r *components) renderComponent(ctx context.Context, acc v1alpha2.Applicati
 			return nil, err
 		}
 	} else {
-		// we have a completely different approach on workload name for application generated appConfig
-		revision, err := utils.ExtractRevision(acc.RevisionName)
-		if err != nil {
-			return nil, err
-		}
-		SetAppWorkloadInstanceName(acc.ComponentName, w, revision)
-		if isComponentRolling && needRolloutTemplate {
-			// we have a special logic to emit the workload as a template so that the rollout
-			// controller can take over.
-			// TODO: We might need to add the owner reference to the existing object in case the resource
-			// is going to be shared (ie. CloneSet)
-			if err := prepWorkloadInstanceForRollout(w); err != nil {
+		// we have completely different approaches on workload name for application generated appConfig
+		if c.Spec.Helm != nil {
```

Comment:
why do we even allow helm-related logic in AC? IIRC, the application controller directly applies the helm chart, it doesn't go through AC.

Reply:
It's used to apply traits to the workload. Then, workload from Helm chart can also enjoy the whole trait system.